### DISCUSSION TITLE
ccache with Tasking compiler

### DISCUSSION THREAD

**Discussion by wils93** [ccache/ccache#1106] [Support]
Hi 👋
When trying to integrate ccache with Tasking compiler, we get "**Multiple source files**" due to the fact that cctc.exe has -f <file_with_all_include_paths>. So is it possible to have this up and working?

example of compilation command:
`cctc.exe <compiler_options> -f <file_with_all_included_paths> <file.c>`

**Comment by jrosdahl** [comment]
Hi,

Do you mean that you want to use ccache with the Tasking TriCore C/C++ compiler? Ccache currently doesn't support that. I've added a note about it on _[Supported platforms, compilers and languages](https://ccache.dev/platform-compiler-language-support.html)_ now. See also #841 and #942.

**Reply by wils93** [reply]
> Hi,
> 
> Do you mean that you want to use ccache with the Tasking TriCore C/C++ compiler? Ccache currently doesn't support that. I've added a note about it on _[Supported platforms, compilers and languages](https://ccache.dev/platform-compiler-language-support.html)_ now. See also #841 and #942.

Hi,

I was trying Tasking compiler as being listed as an EDG compiler.
Is there a plan if/when it will be supported?

**Reply by jrosdahl** [reply]
> I was trying Tasking compiler as being listed as an EDG compiler.

I see. I didn't know that Tasking Tricore was EDG-based. I've relaxed the statement on _[Supported platforms, compilers and languages](https://ccache.dev/platform-compiler-language-support.html) now.

> Is there a plan if/when it will be supported?

No. As mentioned in https://github.com/ccache/ccache/pull/942#issuecomment-958124266 and
https://github.com/ccache/ccache/pull/942#issuecomment-1040731590 it depends on whether somebody has time and interest to work on a compiler framework (#956) and preferably a better testing framework (#957).
